index.bs

```diff
@@ -1765,6 +1792,39 @@ a [=moment=] |sourceTime|, and a [=duration=] |eventReportWindow|:
     1. Set |lastEnd| to |lastEnd| + |deadline|.
 1. Return |windows|.
 
+To <dfn>parse report windows</dfn> given a [=map=] |map|, a [=moment=] |sourceTime|,
+and an [=report window list=] |default|:
+
+1. [=Assert=]: |default| is not [=list/is empty|empty=].
+1. Let |defaultEndDuration| be the [=duration from=] |sourceTime| and the [=report window/end=] of the last element of |default|.
+1. If |map|["`event_report_windows`"] does not [=map/exist=], return |default|.
+1. Let |values| be |map|["`event_report_windows`"].
+1. If |values| is not a [=map=], return an error.
+1. Let |startDuration| be 0 seconds.
+1. If |values|["`end_times`"] does not [=map/exist=] or is not a [=list=], return an error.
+1. Let |endDurations| be |values|["`end_times`"].
+1. If the [=list/size=] of |endDurations| is greater than [=max settable event-level report windows=], return an error.
+1. If |endDurations| [=list/is empty=], return an error.
+1. Let |windows| be an [=list/is empty|empty=] [=list=].
+1. [=list/iterate|For each=] |elt| of |endDurations|:
+    1. If |elt| is not a [=string=], return an error.
+    1. Let |end| be the result of applying the <a spec="html">rules for parsing integers</a> to |elt|.
+    1. If |end| is an error, return an error.
+    1. Let |endDuration| be |end| seconds.
+    1. If |endDuration| is greater than |defaultEndDuration|, set |endDuration| to |defaultEndDuration|.
+    1. If |endDuration| is less than 1 hour, set |endDuration| to 1 hour.
```

Comment:
Instead of 1 hour, should this be referencing https://wicg.github.io/attribution-reporting-api/#min-report-window?